padl/transforms.py

```diff
@@ -526,17 +526,15 @@ def f(x):
 
             # see if the object itself knows how to generate its codegraph
             try:
-                if next_name.scope.is_global():
-                    next_obj = globals_dict[next_name.name]
-                else:
-                    next_obj = all_vars_dict[next_name.name]
+                next_obj = all_vars_dict[next_name.name]
```

Comment:
@wuhu This is a change @jasonkhadka and I discussed since it looks like `all_vars_dict` already contains the `globals_dict`. Was there any other reason to have this `if` statement?

Reply:
Okay thanks for the info! @jasonkhadka Could we add the example provided by @wuhu to the unit tests to test this scenario? Also I think it would be helpful to add a comment in the code explaining this situation or at least that `all_vars_dict` relies on `nonlocals_dict` overwriting the keys in `globals_dict`.

Reply:
Then the question becomes if there are cases when the scope is not global for an object but it is in `globals_dict`. I think this can be the case when we are looking at some internal scope, where variables from the outer scope are used, but not necessarily global. 
Is this the case @wuhu? 
If not, then we can skip defining `all_vars_dict`, and use `non_local_dict` in `else` clause.

Reply:
Thanks for the clarifaction. I reverted this change.